adoc-mode.el

```diff
@@ -248,6 +257,48 @@ See for example `tempo-template-adoc-title-1'."
                  (const :tag "tempo-snippets" tempo-snippets))
   :group 'adoc)
 
+(defcustom adoc-fontify-code-blocks-natively nil
+  "When non-nil, fontify code in code blocks using the native major mode.
+This only works for code blocks where the language is
+specified where we can automatically determine the appropriate
+mode to use.  The language to mode mapping may be customized by
+setting the variable `adoc-code-lang-modes'."
+  :group 'markdown
+  :type 'boolean
+  :safe 'booleanp)
+
+;; This is based on `org-src-lang-modes' from org-src.el
+(defcustom adoc-code-lang-modes
+  '(("ocaml" . tuareg-mode) ("elisp" . emacs-lisp-mode) ("ditaa" . artist-mode)
```

Comment:
I'd suggest having one entry per line and having this sorted by language name, so it's easier to scan by humans.

Reply:
CPE.

Reply:
I did re-order the entries.